influxql/select.go

```diff
@@ -822,18 +822,40 @@ func buildTransformIterator(lhs Iterator, rhs Iterator, op Token, ic IteratorCre
 			left:  newBufIntegerIterator(left),
 			right: newBufIntegerIterator(right),
 			fn: func(a *IntegerPoint, b *IntegerPoint) *FloatPoint {
-				p := &FloatPoint{
-					Name: a.Name,
-					Tags: a.Tags,
-					Time: a.Time,
-					Aux:  a.Aux,
-				}
-				if (a != nil && b != nil) && (!a.Nil && !b.Nil) {
-					p.Value = fn(a.Value, b.Value)
+				if a != nil && b != nil {
```

Comment:
if a == nil {
    return &FloatPoint{
        Name: b.Name, //this will be panic if b is nil.
    }
} else if b == nil {
    return &FloatPoint{.....}
}